src/simplify_algebra.cpp

```diff
@@ -421,9 +439,18 @@ struct find_splits
                 if(it == split->outputs().end())
                     break;
                 assert((*it)->name() != "slice");
-                // If there is a duplicate bail
+
                 if(contains(group, *it))
                     return {};
+
+                // there are should be no dependent between instructions in the group
+                if(std::any_of(group.begin(), group.end(), [&](auto i) {
+                       return is_dependent(*it, i) or is_dependent(i, *it);
```

Comment:
Can't the `contains(group, *it)` be removed?

Reply:
removed